lightningd/lightningd.c

```diff
@@ -180,47 +280,85 @@ static const char *find_my_path(const tal_t *ctx, const char *argv0)
 			errx(1, "Cannot find %s in $PATH", argv0);
 	}
 
+	/*~ The caller just wants the directory we're in.
+	 *
+	 * Note the magic "take()" macro here: it annotates a pointer as "to
+	 * be taken", and the recipient is expected to take ownership of the
+	 * pointer.
+	 *
+	 * Many CCAN and our own routines support this, but if you hand a take()
+	 * to a non-take routine unfortunately you don't get a compile error.
+	 */
 	return path_dirname(ctx, take(me));
 }
+
+/*~ This returns the PKGLIBEXEC path which is where binaries get installed.
+ * Note the 'TAKES' annotation which is merely documentation that it will
+ * take ownership of 'my_path' if the caller hands take() there.
```

Comment:
ownership means that it is a node of the parent tal pointer and that my_path can now be freed withour resulting in seg fault ?

Reply:
and: how are take(...) variable and argument TAKES connected? It must be possible to call take more than once within one function?

Reply:
Enhanced, I hope I covered that now.